### ISSUE TITLE
test_export_best

### ISSUE THREAD

**Issue by Anthchirp** [dials/dials#782]
Test currently has two issues
1. it fails for no apparent reason
2. it also runs import, find spots, index, integrate. do we really need all those steps or should we rather use the existing experiments.json instead?

**Comment by Anthchirp** [comment]
With https://github.com/cctbx/dxtbx/commit/dca41d4ca57d7057c63ee26b227365c3b76d62e5 we now get a more expressive error message.

It does neither explain how this could happen nor why rerunning the exact same command later just works. Could this be a case of a weird exception being thrown, caught and eaten somewhere before?

```
(..)
Timing information for integration
 --------------------------------
         Read time | 1.89 seconds
      Extract time | 0.21 seconds
  Pre-process time | 0.00 seconds
      Process time | 4.97 seconds
 Post-process time | 0.00 seconds
        Total time | 8.20 seconds
         User time | 0.00 seconds
 --------------------------------

Saving 1979 reflections to integrated.pickle
 time taken: 0.010916
Saving the experiments to integrated_experiments.json
 time taken: 0.00816798

Total time taken: 12.457199
DIALS (2018) Acta Cryst. D74, 85-97. https://doi.org/10.1107/S2059798317017235
DIALS 2.dev
The following parameters have been modified:

format = mtz sadabs nxs mmcif mosflm xds *best xds_ascii json
input {
  experiments = integrated_experiments.json
  reflections = integrated.pickle
}

Selecting 1979/1979 reflections with partiality >= 0.1
Traceback (most recent call last):
  File "/Users/dlshudson/jenkins_slave/workspace/dials_regression/label/dials-mac-mini/build_dials/build/../modules/dials/command_line/export.py", line 718, in <module>
    exporter.export()
  File "/Users/dlshudson/jenkins_slave/workspace/dials_regression/label/dials-mac-mini/build_dials/build/../modules/dials/command_line/export.py", line 568, in export
    "%s.dat" % prefix, imageset, n_bins=self.params.best.n_bins
  File "/Users/dlshudson/jenkins_slave/workspace/dials_regression/label/dials-mac-mini/build_dials/modules/dials/util/best.py", line 13, in write_background_file
    d, I, sig = background(imageset, imageset.indices()[0], n_bins=n_bins)
  File "/Users/dlshudson/jenkins_slave/workspace/dials_regression/label/dials-mac-mini/build_dials/modules/dials/command_line/background.py", line 134, in background
    mask = mask_generator.generate(imageset)
  File "/Users/dlshudson/jenkins_slave/workspace/dials_regression/label/dials-mac-mini/build_dials/modules/dials/util/masking.py", line 157, in generate
    image = imageset.get_raw_data(0)
RuntimeError: dxtbx Error: /Users/dlshudson/jenkins_slave/workspace/dials_regression/label/dials-mac-mini/build_dials/modules/dxtbx/imageset.h(518): Unknown type NoneType
```

**Comment by Anthchirp** [comment]
Part 1 is now fixed by 79cf1ad6db796e2e0a443b2694845aa5c6f0ac44 _and_ af3d32b74ae6451330af7a910d384db7b3d047cb.

However I still think the test should just run export, not integrate.
